service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AddNoteController.java

```diff
@@ -26,21 +25,22 @@
 public class AddNoteController {
     private final CaseNoteService service;
     private final RequestData requestData;
-    private final ObjectMapper mapper;
+    private final CaseDataConverter caseDataConverter;
```

Comment:
if we are looking for generic solution then we could have abstract controller which is extended by all our controlers. It can have e.g method saveCase(CaseData caseData){
 return AboutToStartOrSubmitCallbackResponse.builder()
            .data(caseDataConverter.convertToMap(updatedCase))
            .build();
}

then our controller at the end just invoke saveCase(CaseData caseData)

Reply:
I like this too but maybe i will try to leave current scope to just removing strings.